wiring/inc/spark_wiring_cellular_printable.h

```diff
@@ -58,6 +51,13 @@ class CellularSignal : public particle::Signal, public Printable {
 
     virtual size_t printTo(Print& p) const;
 
+    virtual bool isAccessTechnologyValid() const;
+    virtual bool isSignalStrengthValid() const;
+    virtual bool isSignalQualityValid() const;
+    virtual bool isSignalQualitySupported() const;
+    virtual bool isValid() const;
+    virtual operator bool() const;
```

Comment:
No strong opinion here, but I'd prefer to have either `isValid()` or a bunch of methods like `isStrengthValid()`, etc, but not both.

Reply:
It looks like isValid() was left and the individual methods were removed.  Since we also have the bool return which is basically a duplication of isValid(), would it be more useful to users to have the individual isX() methods?  Especially when the bool return is false, they will have a built in way to determine what has gone wrong.  This can also be very useful for internal testing.

Reply:
@avtolstoy @sergeuz @keeramis any opinions on if we should bring back the isX() methods or leave them out?  My reasoning on adding them back in is users would have to add special code that's not really going to be well documented IMO, or if we documented it it would be dense at best.  Or we'd have to document an exact example for validation code. For example, they would have to know they should qualify the value with `sig_.rat != NET_ACCESS_TECHNOLOGY_NONE && value !=>< x` after looking up what the default/error values are for each API.

Reply:
No strong opinions, but I'd probably try to keep things simpler right now and if needed when we get feedback later introduce distinct methods.

However, in general the whole object should be valid and there shouldn't be cases where a particular part of it is invalid. Understandably we do have a quirk on one of the platforms with a particular modem type (which is also documented https://docs.particle.io/reference/device-os/firmware/boron/#getqualityvalue-), but taking all the other platforms into consideration a simpler approach seems more desirable. Thoughts?